pandas/core/arrays/_arrow_string_mixins.py

```diff
@@ -49,7 +53,19 @@ def _str_pad(
         elif side == "right":
             pa_pad = pc.utf8_rpad
         elif side == "both":
-            pa_pad = pc.utf8_center
+            if pa_version_under17p0:
+                # GH#59624 fall back to object dtype
+                from pandas import array
+
+                obj_arr = self.astype(object, copy=False)  # type: ignore[attr-defined]
+                obj = array(obj_arr, dtype=object)
```

Comment:
Could also do `obj_arr.array._str_pad(..)` and then don't need this explicit construction?